### PR TITLE
Issue #16 skipped functionality

### LINKED ISSUE #16: Skip: Show Ignored in Output?
**Note: it might not be possible with how. we use `Deno.test`. Maybe we could add custom output outselves to say it was skipped (but it might still show as OK), or (and i really dont want to do this), change our `Deno.test` signature to `Deno.test({name: ..., ignore: ..., async fn() ...})`**

## Summary

What: Modify the `skip` method in `RhumRunner` so it stills calls `Deno.test`, but `ignore: true` is passed in. This would probably require a change to `ITestCase` to set if a test should be ignored. Defaults to false

Why: I think it would be good that the user would see the test is being ignored (like now `deno test` currently displays it)

## Acceptance Criteria

- Skipped tests would show as ignored in the output (all tests inside a skipped suite, all tests inside a skipped plan, etc)
- Unit tests for `.skip` method
- Integration test for using skip: `tests/integration/skip.ts`

## Example Pseudo Code (for implementation)

```typescript
// run()

Deno.test(..., { ignore: test.ignore })

// skip
public skip (name, fn) {
  if (this.is_a_test) {
    this.tests.push(..., ignore: true)
  else if this.passed_in_test_suite // skipped a suite
    this.skipped_suites.push(name)
  else // plan
    this.skip_plan = true

// somewhere else
this.plan.suites.foreach
  suite.cases.foreach
    if this.skipped_suites.indexOf(suite.name)
      case.ignore =. true
    if this.skip_plan === true
      cas.eignore = true
```

### PR DESCRIPTION
**Pull request by crookse**
Fixes #16 

**Description**

`.testSuite()` and `.testCase()` calls can be replaced with `.skip()`.

You can now do the following:

```typescript
Rhum.testPlan(() => {
  Rhum.skip("testSuite skipped 1", () => {
    Rhum.testCase("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.testCase("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.skip("testSuite skipped 2", () => {
    Rhum.testCase("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.testSuite("testSuite not skipped 3", () => {
    Rhum.testCase("testCase", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.testSuite("testSuite not skipped 4", () => {
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.testCase("testCase", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.testSuite("testSuite not skipped 5", () => {
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.testCase("testCase", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.skip("testSuite skipped 6", () => {
    Rhum.testCase("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });

  Rhum.testSuite("testSuite not skipped 7", () => {
    Rhum.testCase("testCase", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.skip("skipped", () => {
      Rhum.asserts.assertEquals(true, true);
    });
    Rhum.testCase("testCase", () => {
      Rhum.asserts.assertEquals(true, true);
    });
  });
});

```

### CODE REVIEW

File: mod.ts [hunk 1]
```diff
@@ -226,10 +311,51 @@ export class RhumRunner {
...
+
+  /**
+   * Skip a test case.
+   *
+   * @param name - The name of the test case.
+   * @param testFn - The test to execute.
+   */
+  public async skipTestCase(name: string, testFn: () => void): Promise<void> {
+    this.addTestCaseToTestSuite(name, testFn, true, this.getCurrentTestSuite());
+  }
```
Comment (ebebbington) [inline mod.ts:342]: Async fn but doesn't seem to do anything async?

**Review by ebebbington — COMMENTED** [review]
I think maybe some more tests could be added, mainly integration ones that assert the output is correct with the new format

**Comment by crookse** [comment]
> I think maybe some more tests could be added, mainly integration ones that assert the output is correct with the new format

agreed. as i'm testing the rhumrunner, i'm finding that some of my code isn't testable. so i'm also reworking some of the stuff i did to make the rhumrunner testable as fuhhhh

File: mod.ts (hunk 1 above)
Comment (crookse) [reply mod.ts]: 15f7b796cdf

**Comment by crookse** [comment]
@ebebbington, i added out `output_test.ts`. i also tested `mod.ts`.

https://github.com/drashland/rhum/blob/issue-%2316-skipped-functionality/tests/integration/output_test.ts

https://github.com/drashland/rhum/blob/issue-%2316-skipped-functionality/tests/unit/mod_test.ts

**Review by ebebbington — APPROVED** [review]
LGTM

**Pull request merged**